nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -22,12 +22,18 @@ let
 
   kernel = config.boot.kernelPackages;
 
-  splKernelPkg = kernel.spl;
-  zfsKernelPkg = kernel.zfs;
-  zfsUserPkg = pkgs.zfs;
+  packages = if lib.versionAtLeast kernel.kernel.version "4.9" then {
```

Comment:
This is indeed questionable.
I will replace it by an assert and a feature flag.